From e481b1f8d17364e71357695fa13bc6982419ccb3 Mon Sep 17 00:00:00 2001
From: denglei <denglei@eswincomputing.com>
Date: Fri, 24 Oct 2025 10:33:17 +0800
Subject: [PATCH 533/540] fix:LLC dts node add TBU number.

Changelogs:
	Resolve the error caused by the LLC driver during system resume.

Signed-off-by: denglei <denglei@eswincomputing.com>
---
 .../boot/dts/eswin/eswin-win2030-die0-soc.dtsi      |  1 +
 drivers/interconnect/eswin/noc.c                    | 13 +++----------
 sound/soc/eswin/esw-i2s.c                           |  2 +-
 3 files changed, 5 insertions(+), 11 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
index 26857746de42..02f5ded734f4 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
@@ -622,6 +622,7 @@ dev_llc_d0: llc@51c00000 {
 				<&d0_reset NPU_RST_CTRL SW_NPU_LLC_RSTN>;
 			reset-names = "axi", "cfg", "core", "llc";
 			numa-node-id = <0>;
+			tbus = <WIN2030_TBUID_NPU>;
 			spram-region = <&npu0_reserved>;
 		};
 
diff --git a/drivers/interconnect/eswin/noc.c b/drivers/interconnect/eswin/noc.c
index c6d44719db36..078819cec4df 100644
--- a/drivers/interconnect/eswin/noc.c
+++ b/drivers/interconnect/eswin/noc.c
@@ -2316,7 +2316,7 @@ static int win2030_noc_remove(struct platform_device *pdev)
 	kfree(noc_device->prof);
 	return 0;
 }
-#ifdef CONFIG_PM
+
 static int noc_resume(struct device *dev)
 {
 	struct win2030_noc_device *noc_device = dev_get_drvdata(dev);
@@ -2347,13 +2347,8 @@ static int noc_suspend(struct device *dev)
 
 	return 0;
 }
-static const struct dev_pm_ops win2030_noc_driver_pm_ops = {
 
-	.suspend = noc_suspend,
-	.resume = noc_resume,
-
-};
-#endif
+static DEFINE_SIMPLE_DEV_PM_OPS(win2030_noc_driver_pm_ops, noc_suspend, noc_resume);
 
 static struct platform_driver win2030_noc_driver = {
 	.probe = win2030_noc_probe,
@@ -2361,9 +2356,7 @@ static struct platform_driver win2030_noc_driver = {
 	.driver = {
 		.name = "win2030-noc",
 		.owner = THIS_MODULE,
-#ifdef CONFIG_PM
-		.pm = &win2030_noc_driver_pm_ops,
-#endif
+		.pm = pm_sleep_ptr(&win2030_noc_driver_pm_ops),
 		.of_match_table = of_match_ptr(win2030_noc_of_match),},
 };
 
diff --git a/sound/soc/eswin/esw-i2s.c b/sound/soc/eswin/esw-i2s.c
index 9f56fd7a9223..1f44ff61d1fd 100755
--- a/sound/soc/eswin/esw-i2s.c
+++ b/sound/soc/eswin/esw-i2s.c
@@ -1187,7 +1187,7 @@ static struct platform_driver i2s_driver = {
 	.driver		= {
 		.name	= "es-i2s",
 		.of_match_table = of_match_ptr(i2s_of_match),
-		.pm = pm_ptr(&i2s_pm_ops),
+		.pm = pm_sleep_ptr(&i2s_pm_ops),
 	},
 };
 
-- 
2.54.0

